testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/AgentTestRunner.java

```diff
@@ -109,9 +112,17 @@
         .getClass()
         .getSimpleName()
         .equals("NoopTextMapPropagator")) {
-      OpenTelemetry.setGlobalPropagators(
-          DefaultContextPropagators.builder()
-              .addTextMapPropagator(HttpTraceContext.getInstance())
+      // Workaround https://github.com/open-telemetry/opentelemetry-java/pull/2096
+      OpenTelemetry.set(
+          OpenTelemetrySdk.builder()
+              .setResource(OpenTelemetrySdk.get().getResource())
+              .setClock(OpenTelemetrySdk.get().getClock())
+              .setMeterProvider(OpenTelemetry.getGlobalMeterProvider())
+              .setTracerProvider(unobfuscate(OpenTelemetry.getGlobalTracerProvider()))
+              .setPropagators(
+                  DefaultContextPropagators.builder()
+                      .addTextMapPropagator(HttpTraceContext.getInstance())
+                      .build())
               .build());
```

Comment:
Minor reuse suggestion:

```suggestion
      setGlobalPropagators(DefaultContextPropagators.builder()
              .addTextMapPropagator(HttpTraceContext.getInstance())
              .build());
```